src/cladetime/util/reference.py

```diff
@@ -153,3 +162,66 @@ def get_nextclade_dataset(
     _run_nextclade_cli(nextclade_cli_version, command, output_file)
 
     return output_file
+
+
+def get_clade_assignments(
+    nextclade_cli_version: str, sequence_file: Path, nextclade_dataset: Path, output_path: Path
+) -> Path:
+    """Assign clades to sequences using the Nextclade CLI.
+
+    Run the Nextclade CLI :external:doc:`dataset run<user/nextclade-cli/usage>`
+    command and save the resulting clade assignment file to disk. The clade
+    assignment file will be in CSV format.
+
+    Parameters
+    ----------
+    nextclade_cli_version : str
+        Version of the Nextclade CLI to use. Used as the tag when
+        pulling the Nextclade CLI Docker image (e.g., "3.8.2")
+    sequence_file : pathlib.Path
+        Location of the sequence file to assign clades to. The file should
+        be in FASTA format.
+    nextclade_dataset : pathlib.Path
+        Location of the :external:doc:`Nextclade dataset<user/datasets>`
+        that contains the reference tree and root sequence to use
+        for clade assignment. Use :func:`get_nextclade_dataset` to
+        get a dataset that corresponds to a specific point in time.
+    output_path : pathlib.Path
+        Where to save the clade assignment file
+
+    Returns
+    -------
+    pathlib.Path
+        Full path to the
+        :external:doc:`clade assignment file`<user/output-files/04-results-tsv>
+        created by Nextclade
+
+    Raises
+    ------
+    NextcladeNotAvailableError
+        If there is an error creating a Docker client or running Nextclade
+        CLI commands using the Docker image.
+    """
+    assignment_filename = "nextclade_assignment.csv"
+    output_file = output_path / assignment_filename
+    output_path.parent.mkdir(parents=True, exist_ok=True)
```

Comment:
I'm finding the management of `output_file` and `output_path` to be a little involved to think through:
- In this function, we accept `output_path`, a directory where outputs will be saved, as an argument
- We construct `output_file`, a file with name we specify living in `output_path`
- We then pass `output_file` to `_run_nextclade_cli`
- The only thing `_run_nextclade_cli` does with that is extract the `output_path` to use for a volume mount
- `_run_nextclade_cli` then returns `output_file`, which we ignore here because we already knew that (we passed as an argument in our `_run_nextclade_cli` call!)
- we also passed `        f"/data/{nextclade_dataset.name}",` as part of our commands, which relies on knowing that `_run_nextclade_cli` was going to mount `output_path` as `/data/`.  Probably no way around this, but it added a point of confusion for me reading this through.

Could we simplify by doing these things?
 - passing `output_path` to `_run_nextclade_cli` rather than `output_file`
 - not bothering to return `output_file` from `_run_nextclade_cli`

Reply:
Agree we should clean this up. I'd prefer to make sure we can get an end-to-end process in place for creating target data first (i.e., unblock people), so I filed an issue: https://github.com/reichlab/cladetime/issues/52